### DISCUSSION TITLE
Show one of the field of the child layer in the parent layer form

### DISCUSSION THREAD

**Discussion by Plant15** [opengisch/QField#2933] [Q&A]
Hi,

For my vegetation surveys I plan to use QField with a child layer allowing me to note multiple species for one parent spatial layer. Is it possible to show, for example, the field for the species ("arbre_sp") in the parent layer widget list? For now its blank or it has the uuid relation value between my parent and child layer. Thank you

![Screenshot_20220524-105604](https://user-images.githubusercontent.com/98606616/170067042-82cddeba-e85d-4c55-a235-aefb3a49ba8f.png)

![Screenshot_20220524-104314](https://user-images.githubusercontent.com/98606616/170067016-e76a1ca4-7f98-4665-8c40-f0a473083575.png)

![Screenshot_20220524-104306](https://user-images.githubusercontent.com/98606616/170067036-fe53718a-25dc-4479-a242-174987db4b3f.png)

**Comment by OBVYamaska** [comment]
We have a similar form using Relation values to get species from a second layer with all our vegetation database.
You can personalize the title that appears in the list when you are in the form view of that section, in QGIS. Click on the button that shows "Expression", and you'll get various options to name it, either using a pre-prepared expression or building your own. We do this to show the absolute and relative coverage of each species in a wetland or riparian strip form, for instance.
![image](https://user-images.githubusercontent.com/101751953/170089452-dfeb71e6-1580-4e74-9026-0106eaac1066.png)

**Reply by Plant15** [reply]
That's exactly what I was looking for. Thank you! However his functionality seems to be applicable only when the relation is "simple" i.e not polymorphic. A polymorphic relation in which many parent layer have a relation to a single table of "ally species" allow edition of new species but these are not linked to the parent layer. What am I doing wrong? It would be long to create 3 "simple" relations for every growth form (10) in my vegetation form.


![image](https://user-images.githubusercontent.com/98606616/170140933-4e8a835f-61c5-4732-aa80-31a0ca623555.png)

![image](https://user-images.githubusercontent.com/98606616/170141334-4bf5d122-77d4-4c2d-8170-4dab10203fbc.png)

**Reply by OBVYamaska** [reply]
I'm not very familiar with polymorphic relations, but... are they necessary for your case? Couldn't you simply add  multiple regular "species" forms to a single "Observation/Site" form? This is what we do and it works great.
Observation layer, linked to a Caracterisation form, which uses Value Relation widgets to seek data in a Species table.
![image](https://user-images.githubusercontent.com/101751953/170355158-ba458887-ccaa-43a2-add5-2b528c17a0d7.png)
![image](https://user-images.githubusercontent.com/101751953/170355211-ccad21a7-a28b-4c80-99ab-53bdfc14f2d3.png)
![image](https://user-images.githubusercontent.com/101751953/170357224-30960818-ecd1-4598-b996-53764c1d602c.png)



P.S: For the record, I continue in english fot the sake of documenting the idea in the community, even though we are clearly both French speaking ;)

**Comment by Plant15** [comment]
Thats ok, I will do my 30 entries that way instead. Thank you. Yes, odds were truly low that two Quebeckers would answers each other here in english. Good continuation ;)

**Reply by OBVYamaska** [reply]
It seems like we are doing similar projects on the field, so if you're interested, it could be fun to share our methods and maybe learn ways to do it with QField from each other!

**Reply by MatparMcWolfie** [reply]
If you'd allow another Quebecker to chime in (two years late ...), I've also created similar forms, most notably one to perform complete wetland surveys as per Lachance et al. method, along with full auto-filled field reports. If you're interested in sharing ideas and methods, let me know!

**Reply by OBVYamaska** [reply]
@MatparMcWolfie , I would like to see how you approached the subject! We've been working on these forms for some time now and are sharing them with a lot of partners (See the Geofluence initiative). I'm not exactly sure what's the best way to get in contact from Github... but I invite you to write me an email at michel.landry [at] obv-yamaska.qc.ca

**Reply by nirvn** [reply]
Third quebecois chiming in here :wink: 

The name used in the list of children can be modified through editing the children layer's display name property in the vector layer dialog, here:

![image](https://github.com/user-attachments/assets/d9302424-3de6-4a1f-8918-d3acdc419dc2)

**Reply by MatparMcWolfie** [reply]
> @MatparMcWolfie , I would like to see how you approached the subject! We've been working on these forms for some time now and are sharing them with a lot of partners (See the Geofluence initiative). I'm not exactly sure what's the best way to get in contact from Github... but I invite you to write me an email at michel.landry [at] obv-yamaska.qc.ca

I'll get in touch for sure!
